rascal-vscode-extension/src/extension.ts

```diff
@@ -68,94 +78,39 @@ function activateTerminal(context: vscode.ExtensionContext) {
-			name: 'Rascal Terminal',
+		const reply:Promise<IDEServicesConfiguration> = client.sendRequest("rascal/supplyIDEServicesConfiguration");
+
+		reply.then((cfg:IDEServicesConfiguration) => {
+			let terminal = vscode.window.createTerminal({
+				cwd: path.dirname(uri.fsPath),
+				shellPath: getJavaExecutable(),
+				shellArgs: [
+					'-cp' , buildJVMPath(context), 
+					'org.rascalmpl.vscode.lsp.terminal.LSPTerminalREPL',
+					'--ideServicesPort', 
+					'' + cfg.port
+				],
+				name: 'Rascal Terminal',
+			});
+
+			context.subscriptions.push(disposable);
+			terminal.show(false);
 		});
-
-		registerContentViewSupport();
-		context.subscriptions.push(disposable);
-		terminal.show(false);
 	});
 }
 
-function registerContentViewSupport() {
-	vscode.window.registerTerminalLinkProvider({
-		provideTerminalLinks: (context, token) => {
-			var pattern: RegExp = new RegExp('Serving \'(?<theTitle>[^\']+)\' at \\|http://localhost:(?<thePort>[0-9]+)/\\|');
-			var result:RegExpExecArray = pattern.exec(context.line)!;
-
-			if (result !== null) {
-				let port = result.groups!.thePort;
-				let matchAt = result.index;
-				let title = result.groups!.theTitle;
-
-				return [
-					{
-						startIndex: matchAt,
-						length: result?.input.length,
-						tooltip: 'Click to view ' + title,
-						url: 'http://localhost:' + port + '/',
-						contentType: 'text/html',
-						contentId: title
-					}
-				];
-			}	
-		  
-			return [];
-		},
-		handleTerminalLink: (link: vscode.TerminalLink) => {
-			let theLink = (link as RascalTerminalContentLink);
-			let url = theLink.url;
-			let oldPanel:vscode.WebviewPanel = contentPanels.find(p => (p as vscode.WebviewPanel).title === theLink.contentId);
-
-			if (oldPanel === undefined) {
-				const panel = vscode.window.createWebviewPanel(
-					theLink.contentType,
-					theLink.contentId,
-					vscode.ViewColumn.One,
-					{
-						enableScripts: true,
-					}
-				);
-
-				contentPanels.push(panel);
-				panel.webview.html = `
-				<!DOCTYPE html>
-				<html lang="en">
-				<head>
-					<meta charset="UTF-8">
-					<meta name="viewport" content="width=device-width, initial-scale=1.0">
-				</head>
-				<body>
-				<iframe id="iframe-rascal-content" src="${url}" frameborder="0" sandbox="allow-scripts allow-forms allow-same-origin allow-pointer-lock allow-downloads allow-top-navigation" style="display: block; margin: 0px; overflow: hidden; position: absolute; width: 100%; height: 100%; visibility: visible;">
-				Loading ${theLink.contentId}...
-				</iframe>
-				</body>
-				</html>`;
-
-				panel.onDidDispose((e) => {
-					contentPanels.splice(contentPanels.indexOf(panel), 1);
-				});
-			} else {
-				oldPanel.reveal(vscode.ViewColumn.One, false);
-			}
-		}
-	});
+function buildJVMPath(context: vscode.ExtensionContext) :string {
+	const jars = ['rascal-lsp.jar', 'rascal.jar', 'rascal-core.jar', 'typepal.jar'];
+	return jars.map(j => context.asAbsolutePath('./dist/' + j)).join(path.delimiter);
```

Comment:
The code has to use ` path.join(` instead of string concat so it also works on windows (which has `\` separated paths)

Reply:
ok